web/response_writer_wrapper.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package web
+
+import (
+	"bufio"
+	"errors"
+	"net"
+	"net/http"
+)
+
+type ResponseWriterWrapper struct {
+	http.ResponseWriter
+	statusCode        int
+	statusCodeWritten bool
+	hijacker          http.Hijacker
```

Comment:
Is there a need to store the hijacker and the flusher ? From what I see here: https://github.com/urfave/negroni/blob/f4316798d5d3acd39eb6784301b19f27f471415f/response_writer.go#L31, we can just cast the response to the appropriate interface and get the job done.

Reply:
Yes, given that we're storing the interface for `http.ResponseWriter` either the `Flush` and `Hijack` methods are not available from the wrapper so you have to get them from the original writer as you can see in [this play example](https://play.golang.com/p/C-WDdmHvn33). In case you want to have it right you have to do something like [this](https://play.golang.com/p/GA7ISH3Gf8q)

I'm pretty sure the Negroni implementation is not working right in the `Hijack` case

Reply:
Ah yes, the negroni implementation is doing exactly that. What do you think about embedding the Flusher interface too ? It makes it cleaner IMO because we are just composing interfaces and not storing any concrete state.

Reply:
In that case, we have to embed the Hijack interface too. I don't have a strong opinion on this so I can do it without problem

Reply:
Yes, I think that's better.

Reply:
Finally given that we don't want a new Interface we let this part as it is (talked offline)